integration_tests/sdk/setup_integration.py

```diff
@@ -173,12 +173,28 @@ def _setup_relational_data(client: Client, db: RelationalDBResource) -> None:
     _add_missing_artifacts(client, db, existing_table_names)
 
 
-def _setup_s3_data(client: Client, s3: S3Resource):
-    # Find all the objects that already exist.
+def _setup_s3_data(client: Client, s3: S3Resource, integration_name: str) -> None:
+    """
+    Find all the objects that already exist.
+    s3 root direcotory may be added to test-config storage section.
+    storage:
+        test-s3:
+            root_dir: "/foo"
+    """
+
     existing_names = set()
     for object_name in demo_db_tables():
         try:
-            s3.file(object_name, artifact_type=ArtifactType.TABLE, format="parquet")
+            root_dir = "/foo/"
+            config_data = _parse_config_file()
+            if "storage" in config_data:
+                if "root_dir" in config_data["storage"][integration_name]:
+                    root_dir = config_data["storage"][integration_name]["root_dir"]
+                    if not root_dir.endswith("/"):
+                        root_dir += "/"
+            s3.file(
```

Comment:
ahhh I see what happened! Shoot I pointed you to this method instead of `setup_data_integrations()`, which is where this change should go. The `root_dir` should only matter when a connect a resource. Afterwards, all the s3.file() stuff should be relative to that (I think?)